pilot/pkg/model/push_context.go

```diff
@@ -763,6 +770,16 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthPolicies(env *Environment, services []*Service) {
+	for _, svc := range services {
+		ps.AuthPolicies[svc.Hostname] = map[int]*Config{}
```

Comment:
multiple services can have the same hostname, I think this will ignore some?

Reply:
@howardjohn I agree.  My current thinking is to index by cluster name and I am working on implementing/testing in my original PR https://github.com/istio/istio/pull/16296.  But @diemtvu's PR https://github.com/istio/istio/pull/16657 was beginning to diverge from my implementation so I wanted to get things back on track